10_1_graph.js

```diff
@@ -0,0 +1,15 @@
+export function buildGraph(edges) {
+    let graph = Object.create(null);
+    function addEdge(from, to) {
+        if (graph[from] == null) {
+            graph[from] = [to];
+        } else {
+            graph[from].push(to);
+        }
+    }
+    for (let [from, to] of edges) {
+        addEdge(from, to);
+        addEdge(to, from);
```

Comment:
Why do you do it twice? It's the same pair, but in different order. You can go through it both ways later, but you don't need duplicated data in here